transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannelConfig.java

```diff
@@ -297,6 +303,14 @@ public EpollSocketChannelConfig setTcpKeepCntl(int probes) {
         return this;
     }
 
+    /**
+     * Set the {@code TCP_MD5SIG} option on the socket. See {@code linux/tcp.h} for more details.
+     */
+    public EpollSocketChannelConfig setTcpMd5Sig(Map<InetAddress, byte[]> keys) {
+        channel.setTcpMd5Sig(keys);
```

Comment:
It definitely is doable. There is one concern, though, which is that the keys may be set up by a trusted component and then handed of to pipeline which runs untrusted code. If we expose this property, anyone who has access to the channel can get the keys. Linux does not provide a getsockopt() equivalent, I guess for exactly this reason.

The other inconvenience is that the map contains byte[] -- hence each get would have to duplicate it so we prevent System.arrayCopy() and similar shenanigans. While the implementation does not care (we only need to know the addresses, so we know what to delete), a malicious accessor could put garbage there and confuse an innocent reader.